src/block-ref.ts

```diff
@@ -68,12 +64,15 @@ export function createBlockRefMiddleware({
 
     // perform child request
     log('Performing another request %o', childRequest);
-    const childRes: PendingJsonRpcResponse<Block> = await pify(
-      provider.sendAsync,
-    ).call(provider, childRequest);
-    // copy child response onto original response
-    res.result = childRes.result;
-    res.error = childRes.error;
+    // copy child result onto original response
+    try {
+      const childResult = await provider.request<JsonRpcParams, Block>(
+        childRequest,
+      );
+      res.result = childResult;
+    } catch (error) {
+      res.error = serializeError(error);
+    }
```

Comment:
Hmm, now that I think about this, I don't think we need to catch this error. Because we are inside of an async middleware, if `request` throws, the error will already be caught and included in the response object. So we should just be able to say:

``` suggestion
    // copy child result onto original response
    res.result = await provider.request(childRequest);

Reply:
Yes indeed I removed the catch on retry function but forgot to remove it here